web/src/lib.rs

```diff
@@ -460,6 +462,59 @@ impl Ruffle {
         // Instance is dropped at this point.
     }
 
+    pub fn add_font(&mut self, font_name: &str, data: Uint8Array) {
+        let _ = self.with_core_mut(|core| {
+            let bytes: Vec<u8> = data.to_vec();
+            if let Ok(swf_stream) = swf::decompress_swf(&bytes[..]) {
+                if let Ok(swf) = swf::parse_swf(&swf_stream) {
+                    let encoding = swf::SwfStr::encoding_for_version(swf.header.version());
+                    for tag in swf.tags {
+                        match tag {
+                            swf::Tag::DefineFont(_font) => {
+                                tracing::warn!("DefineFont1 tag is not yet supported by Ruffle, inside font swf {font_name}");
+                            }
+                            swf::Tag::DefineFont2(font) => {
+                                tracing::debug!(
+                                    "Loaded font {} from font swf {font_name}",
+                                    font.name.to_str_lossy(encoding)
+                                );
+                                core.register_device_font(FontDefinition::SwfTag(*font, encoding));
+                            }
+                            swf::Tag::DefineFont4(_font) => {
+                                tracing::warn!("DefineFont4 tag is not yet supported by Ruffle, inside font swf {font_name}");
+                            }
+                            _ => {}
+                        }
+                    }
+                    return;
+                }
+            }
+
+            tracing::warn!("Font source {font_name} was not recognise (not a valid SWF?)");
```

Comment:
Non-blocking, but this feels like a typo recognise -> recognised. P.S.  English is not my primary language.

Reply:
Fixed thank you!